plugins/org.yakindu.sct.generator.genmodel.ui/src/org/yakindu/sct/generator/genmodel/ui/quickfix/SGenQuickfixProvider.java

```diff
@@ -57,7 +72,58 @@ public void apply(EObject element, IModificationContext context) throws Exceptio
 					}
 				});
 	}
+	
+	@Fix(SGenJavaValidator.CODE_REQUIRED_DOMAIN)
+	public void changeToValidDomain(final Issue issue, IssueResolutionAcceptor acceptor) {
+		String[] validDomains = issue.getData()[0].split(";");
+		for(String validDomain : validDomains) {
+			addAcceptor(issue, acceptor, validDomain);
+		}
+	}
+	
+	private void addAcceptor(final Issue issue, IssueResolutionAcceptor acceptor, String validDomain) {
+		acceptor.accept(issue, validDomain, null, null, new ISemanticModification() {
+			@Override
+			public void apply(EObject element, IModificationContext context) throws Exception {
+				if (element instanceof GeneratorEntry) {
+					EObject elementRef = ((GeneratorEntry) element).getElementRef();
+					if (elementRef instanceof Statechart) {
+						TransactionalEditingDomain sharedDomain = DiagramPartitioningUtil.getSharedDomain();
+						AbstractTransactionalCommand refactoringCommand = new AbstractTransactionalCommand(sharedDomain,
```

Comment:
It is also quite a lot of indentations which makes it harder to read. Let's move the command into an own class "DomainChangeCommand"; that is something we might want to reuse somewhere.